pkg/util/uid/uid.go

```diff
@@ -0,0 +1,16 @@
+package uid
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+)
+
+func GenerateK8sValidName(input string) (string, error) {
+	re, _ := regexp.Compile(`[^a-zA-Z0-9-]{1,}`)
+	cleanUp := strings.ToLower(re.ReplaceAllString(input, `-`))
+	if len(cleanUp) > 30 {
+		return "", errors.New("Length of string exceeds 63, invalid for k8s.")
```

Comment:
63 => 30

Reply:
What about commenting to explain we're doing this for the k8s sub-domain limitation?

Reply:
It's actually more of a limitation set by RFC1123 for subdomains in general. This is specific to us really since we combine uid and namespace in the subdomain.

Reply:
Gotcha, okay, will remove.